src/providers/startgg.ts

```diff
@@ -0,0 +1,55 @@
+import { createOAuth2Request, sendTokenRequest } from "../request.js";
+
+import type { OAuth2Tokens } from "../oauth2.js";
+
+const authorizationEndpoint = "https://start.gg/oauth/authoriz";
+const tokenEndpoint = "https://api.start.gg/oauth/access_token";
+const refreshEndpoint = "https://api.start.gg/oauth/refresh";
+
+export class StartGG {
+	private clientId: string;
+	private clientSecret: string;
+	private redirectURI: string;
+
+	constructor(clientId: string, clientSecret: string, redirectURI: string) {
+		this.clientId = clientId;
+		this.clientSecret = clientSecret;
+		this.redirectURI = redirectURI;
+	}
+
+	public createAuthorizationURL(state: string, scopes: string[]): URL {
+		const url = new URL(authorizationEndpoint);
+		url.searchParams.set("response_type", "code");
+		url.searchParams.set("client_id", this.clientId);
+		url.searchParams.set("state", state);
+		url.searchParams.set("scope", scopes.join(" "));
+		url.searchParams.set("redirect_uri", this.redirectURI);
+		return url;
+	}
+
+	public async validateAuthorizationCode(code: string, scopes: string[]): Promise<OAuth2Tokens> {
```

Comment:
Is the scope parameter necessary even for the token request? First time seeing it
@ Zeden19

Reply:
Apparently it is, weird